### ISSUE TITLE
Feature Request: Necromuna (17)

### ISSUE THREAD

**Issue by kurgol** [AppSynergy/shadow-war-armageddon-roster#25]
Hi,

Just wondering what the possibility of adding the Necromuna '17 rules and factions - love the site thanks.

**Comment by AppSynergy** [comment]
Hi there!

It's certainly possible, someone just needs to write the data files for the various gangs.

I don't have much time myself at the moment, but if you fancy writing them I can give you some pointers :)

**Comment by kurgol** [comment]
Happy to help if you can describe the format of the data files and how you want to receive them, that sounds like exactly the sort of job I could do :)

**Comment by kurgol** [comment]
I have had a quick browse through the files in shadow-war-armageddon-roster/src/data/factions/ and noticed that shadow war characters have fewer stats than Necromunda ones, will that matter?

Other than that should I just be following the same layout as the other faction files in the above folder?

**Comment by AppSynergy** [comment]
Yep, it should be as simple as writing new files like [these](https://github.com/AppSynergy/shadow-war-armageddon-roster/blob/develop/src/data/factions/astra_militarum.toml).

Different stats shouldn't be a problem, we'll just need to switch up [this stub](https://github.com/AppSynergy/shadow-war-armageddon-roster/blob/develop/src/data/stats.toml).

Champions should work as specialists, etc. Is anything else different in Newcromunda? I haven't seen the book yet.

**Comment by kurgol** [comment]
there is a wonderful transcript on reddit of all the official rules of the
Necromunda 2017 release, including the expansion books and white dwarf
articles. Even though I own all of the above I use this reference as it's
all in one place.
https://www.reddit.com/r/necromunda/comments/7er2y8/necromunda_underhive_rules_transcript_en/

**Comment by kurgol** [comment]
a character profile in Necromunda includes 3 additional stats:

m, ws, bs, s, t, w, i, a, ld, *cl, will, int*.

**Comment by kurgol** [comment]
I will have a go at what I think looks correct for all the 'munda gangs and try and submit decent ones based on the astra militarum one.

**Comment by AppSynergy** [comment]
I suggest pick one gang to start with, see how that works.

Are you comfortable installing the app? Using the built-in dev server with hot reload is most convenient.

**Comment by kurgol** [comment]
I will give it a bash over the weekend. I am comfortable in a UNIX/Linux
environment and use Github for a lot of work stuff, so hopefully, I'll be
able to figure it out. Will ask for assistance if I get stuck on something.

**Comment by kurgol** [comment]
https://github.com/AppSynergy/shadow-war-armageddon-roster/pull/27

**Comment by kurgol** [comment]
Though I was not able to get the app to build and run locally so this PR is more to get some feedback if I'm going in the right direction, or if I'm making a total mess out of it.

**Comment by AppSynergy** [comment]
Data file looks good, I'll see if I can run it later and give you any
pointers. Cheers!

**Comment by kurgol** [comment]
slight mishap as I'm learning GitHub, pull request will be back shortly (splitting different gangs into separate branches so that you can accept pull requests granularly).

**Comment by kurgol** [comment]
#27 Goliath (replaces #26) and #28 Delaque

**Comment by AppSynergy** [comment]
Looks good! Didn't take much to get the Goliath gang working:

https://github.com/AppSynergy/shadow-war-armageddon-roster/commit/a3d252b3dc3e662aa6a521c1a0070475c58f1858
I've annotated a few of the changes I made.

**Comment by kurgol** [comment]
there was some weirdness in the gang composition that doesn't match the shadow wars template, for example in the astra militarum profile you have:

[size]
min = 3
max = 10
leader_max = 1
specialist_max = 3

The rules for Necromunda gang composition often include a multiplier on reputation and some other strangeness:
eg. from House Goliath, but I believe they are mostly the same. 
- there must be one leader
- the can be no more than 2 champions, plus 1 for every full 10 reputation the gang has
- the total number of gangers must be equal to or higher than total number of other fighters (leaders, juves and champions but not counting hangers-on)
- a fighter can be equipped with a maximum of three weapons. Weapons with unwieldy trait take up two weapon allocations.

I have not addressed any of these with my initial submission, but if you are happy with the current state I will get the other gangs submitted as pull requests similar to the previous 2?

Thanks for the hand-holding!

**Comment by kurgol** [comment]
@AppSynergy there is a genestealer gang that I am prepping now. It is similar but different to the Shadow War Armageddon Genestealer kill team. 

Should I be looking at making some kind of hybrid merged file, or would I be better off just calling the 'munda one house_genestealer.toml to match the other 'munda gangs?

**Comment by AppSynergy** [comment]
Great! 

**Definitely a new file**, and `house_` would work fine as a convention. `necromunda_` may be better, as long as they're all unique and easy to identify.

I've fixed up and merged the last three gangs. 

+ Look out for typos: "grendades" and "wargrear" seem to be particular favorites of yours! I almost kept `Lasogun`. :)
+ Note the difference between `wargear` (stuff a model starts with) and `weapons.wargear` (stuff a model can buy from a category called 'Wargear')

Other than that all good, thanks for taking the time! :)

**Comment by AppSynergy** [comment]
Btw, I also tried out (and fixed!) installing the app from scratch, this should work ok for you now:

````
git clone git@github.com:AppSynergy/shadow-war-armageddon-roster.git
cd shadow-war-armageddon-roster
npm install
npm run dev
````
If you want to try out your 4 gangs so far. Not sure whether to push live sooner, or wait until we have more gangs / other rules accounted for.

**Comment by kurgol** [comment]
I'm hoping to get the rest of the 'munda gangs out this week. I'll go with house_ for now as it's less typing (and in all cases bar Genestealer, actually matches what they call them in the rules).

I apologise for my typing errors, I am dyslexic and although I try my best some do creep through - I particularly liked lasogun too xD

**Comment by AppSynergy** [comment]
No bother, I can proof-read for you. :) I'd normally suggest a spellchecker
for the data files, but 40k has so many made-up words I'm not sure it'd
help!
Even 'wargear' isn't in the dictionary...

**Comment by kurgol** [comment]
I do use Grammarly everywhere :)

**Comment by kurgol** [comment]
final pull request for the last of the current Necromunda gangs has gone out today (Genestealers from white dwarf) all the rest from the core book and gang 1 and gang 2 expansion have been submitted already!

edit: oops it appears chaos have been released in this months wd, ill get them through tomorrow hopefully once I have picked a copy up.

**Comment by kurgol** [comment]
Chaos and Genestealer PR's are in. The chaos one is a bit odd, there are stats for a beast, but it doesn't look like you can buy one, only have one appear in play if a ritual goes wrong so I left them out. It also doesn't appear they have any "gangers".

Should be enough PR's in now for an initial implementation with all the currently supported options from tabletop.

**Comment by kurgol** [comment]
Hey @AppSynergy everything okay? I noticed we were making great headway then everything seems to have stalled - issues with my PR's or just you have more important things/busy?

**Comment by AppSynergy** [comment]
Just forgot, to be honest! :S:S

Let's look at these last few PRs and see if we can't deploy something now.

Do they all work ok, did you get the dev server running in the end?

**Comment by kurgol** [comment]
to be honest im at the business end of my course so I haven't done anything
other than bang out the last 3 factions following hte other format, I
didn't alter the connecting files but it should be trivial to make these
integrated - they are about on par with my previous pr's!

I'm glad you're okay, was worried you'd had a face/bus related interface
issue!

**Comment by AppSynergy** [comment]
Fixed a few of the obvious errors and banged it live, enjoy!

**Comment by kurgol** [comment]
many thanks, been fun to contribute. If I notice any roster errors will be
sure to send some pr's your way :)

**Comment by AppSynergy** [comment]
Great, thanks!

I also implemented a few bits like the 3 item limit (`unwieldy` property
included) and the 20 credit limit for Juves.

If there's any further Necro limitations you think need introducing, you
could take a look at cannotAddFighter() and weaponsAvailable() in the
Builder.vue. You can see most of the restrictions there, from the SWA
rosters or the two above.

https://github.com/AppSynergy/shadow-war-armageddon-roster/blob/develop/src/components/Builder.vue#L242

**Comment by kurgol** [comment]
that's really useful thanks, will have a look through it and once I've
actually played a few more games (And run the roster through a few games)
ill have a better idea if anything is lacking, or typo :)
